### PR TITLE
Tests and updated parameters

### LINKED ISSUE #3: testing
To ensure functionality. The core concepts should be unittested to avoid introducing errors by contributions

### LINKED ISSUE #8: Multiple parallel simset
This currently works by adjusting the 
```
simset.script_name = "main.py"
```
filename in a simset file. However, currently data folders are shared which is annoying if you want to reset like
```
simset clean
```
because all data in all projects are lost.

Does it make sense to have multiple simset files in a single folder? and if so how should we separate data?

### PR DESCRIPTION
**Pull request by hammal**
Fixed some discussed issues.

Specifically, added #3 
- unittests
- flake8 linting
- a basic check that makes sure the following runs without errors
``` bash
simset init
python main.py simulate setup local
python main.py info
./bash_scripts/local_simulation.sh
python main.py info
python main.py process
simset clean --force
```
and a workflow trigger such that these things are tested on each commit to a PR (like this one :)).

Additionally, fixed the data_folder such that its parameterised in the main simset file, i.e. #8.

Finally, did some cleanup removing functionality that was not fully implemented (as the copy command for simset the command line program).

### CODE REVIEW

File: src/simset/initialize.py [hunk 1]
```diff
@@ -77,7 +77,11 @@ def _remove_folder_if_sure(path: str, exclude: List[str] = []):
             for dir_name in dirs:
                 folder_path = os.path.join(root, dir_name)
                 _remove_folder_if_sure(folder_path)
-                os.removedirs(folder_path)
+                # os.removedirs(folder_path)
+        try:
+            os.removedirs(path)
```
Comment (haguettaz) [inline src/simset/initialize.py:82]: Seems to be wrong -> `folder_path`

File: src/simset/initialize.py [hunk 2]
```diff
@@ -107,11 +111,13 @@ def clean(path: str = os.getcwd(), force: bool = False):
...
-    descision = str(
-        input(f'\nAre you sure you want to delete all simulation data?\ny/n\n')
-    ).lower()
-    if descision.lower() == 'yes' or descision == 'y':
-        _remove_folder_if_sure(simset.data_folders[0], ['README.md'])
+    if not force:
+        descision = str(
+            input('\nAre you sure you want to delete all simulation data?\ny/n\n')
+        ).lower()
+    if force or descision.lower() == 'yes' or descision == 'y':
```
Comment (haguettaz) [inline src/simset/initialize.py:118]: `decision`

File: src/simset/templates/main.py.j2 [hunk 3]
```diff
@@ -3,15 +3,18 @@ from typing import Iterable, Dict, Any
...
 
-## simset.data_folders.append(["path1", "path2", "path3"])
-simset.python_interpreter = "python3.10"
+simset.data_folder = os.path.join(os.getcwd(), ".data")
+simset.python_interpreter = "python"
 simset.memory_requirement = 1024
 simset.euler_email = False
 simset.euler_number_of_cores = 1
 simset.euler_wall_time = {'hours': 4, 'minutes': 0}
+simset.memory_requirement = 1024
```
Comment (haguettaz) [inline src/simset/templates/main.py.j2:16]: delete

File: src/simset/initialize.py (hunk 1 above)
Comment (hammal) [reply src/simset/initialize.py:82]: Checked the code again and actually this makes sense.
path variable originates from function argument line 70.

File: src/simset/initialize.py (hunk 2 above)
Comment (hammal) [reply src/simset/initialize.py]: see [latest commit](https://github.com/hammal/simset/pull/9/commits/e9f0ace6de97f816f2180b101ee0a7d2dcb8770f)

File: src/simset/templates/main.py.j2 (hunk 3 above)
Comment (hammal) [reply src/simset/templates/main.py.j2]: see [latest commit](https://github.com/hammal/simset/pull/9/commits/e9f0ace6de97f816f2180b101ee0a7d2dcb8770f)

**Review by sntsnt — APPROVED** [review]
looks fine!

**Pull request merged**